### DISCUSSION TITLE
ISR - Incremental static regeneration with dynamic nextjs content

### DISCUSSION THREAD

**Discussion by cipriancaba** [tailwindlabs/tailwindcss#10694] [General]
Hi,

We are moving from a static generated site of about 3000 pages to ISR to decrease build times and basically obtain the same effect.

Initially, when statically generating the site, we would just do a double build to point tailwind to the dynamically generated pages. Since these pages are fed from a cms there is no way to know ahead of time all the css classes that we would need to generate. This is why double building was the only way.

However, now that all pages will be generated on the fly, there really is no easy way to generate all the possible classes. How do we make JIT work in this situation?

I see nothing in the documentation and I feel this might be a more common scenario as people move towards ssg, ssr, isr, etc

Any help would be more than welcomed.

Thanks,
Ciprian

**Comment by steveninety** [comment]
I've been trying to figure out the same thing. It's weird that there's so little to find on this topic...

How has your transition to ISR turned out w.r.t. tailwind?

**Reply by cipriancaba** [reply]
I just ended up doing the whitelisting and redeploys whenever a CMS change is done. Not the best and not bullet proof unfortunately
